### PR TITLE
Logger singelton to log to EP

### PR DESCRIPTION
**Pull request by ZobelOmry**
## Context

Enabling to send logs to a configurable EP
Pull details (URL and token) from EP, send logs there and additionally log to console

EP PR
https://github.com/strigo/strigo-app/pull/3519

## Tests

- [x] Send logs to logtail
- [x] Log without config (console)
- [x] Config logger with wrong config
- [x] Log in dev - logs to console

## Screenshots

image

### CODE REVIEW

File: services/logger.ts [hunk 1]
```diff
@@ -12,28 +27,52 @@ export const Logger = {
...
+  log(severity: string, message: any, context: {}) {
     try {
+      if (this.url && this.token && !configManager.getConfig().development) {
+        fetch(this.url, {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+            Authorization: `Bearer ${this.token}`
+          },
+          body: JSON.stringify({
```
Comment (v0vs) [inline services/logger.ts:39]: I think that if we're sending json, then we don't to stringify the content

File: services/logger.ts [hunk 2]
```diff
@@ -12,28 +27,52 @@ export const Logger = {
...
+            severity,
+            message,
+            context,
+            dt: new Date().toISOString()
+          })
+        });
+
+        //TODO: do we wanna log to console too? maybe only from a certain severity level? or only for dev?
```
Comment (v0vs) [inline services/logger.ts:47]: Yep. I'd split it to 2 internal methods - logConsole, logRemote based on development flag

File: services/logger.ts [hunk 3]
```diff
@@ -12,28 +27,52 @@ export const Logger = {
...
       const enrichedMessage = `${new Date().toISOString()} - ${message}`;
       console[severity](enrichedMessage, context ? `\n${JSON.stringify(context)}` : "");
     } catch (err) {
       console.log("Logging error:", err);
     }
-  },
+  }
 
-  debug(message, context = {}) {
+  debug(message: any, context = {}) {
```
Comment (v0vs) [inline services/logger.ts:59]:
```suggestion
  debug(message: string, context = {}) {
```

File: services/logger.ts [hunk 4]
```diff
@@ -12,28 +27,52 @@ export const Logger = {
    * @param {String} message the message to convey.
    * @param {Object} context the object to provide as context.
    */
-  log(severity, message, context) {
+  log(severity: string, message: any, context: {}) {
     try {
+      if (this.url && this.token && !configManager.getConfig().development) {
+        fetch(this.url, {
```
Comment (v0vs) [inline services/logger.ts:33]: We'll need to catch the errors, so it won't bubble up to the application - can cause some side effects even if we don't await on the fetch.

File: services/logger.ts (hunk 1 above)
Comment (v0vs) [reply services/logger.ts]: @ZobelOmry Ping

File: src/strigo/index.ts [hunk 5]
```diff
@@ -37,7 +38,39 @@ export namespace Strigo {
     }
   }
 
-  export function setup(data: SDKSetupData) {
+  async function fetchRemoteConfiguration(token: StrigoToken, development: boolean) {
+    try {
+      const configDomain = development ? "http://localhost:3000" : "https://app.strigo.io";
+      const response = await fetch(`${configDomain}/api/internal/academy/v1/config`, {
```
Comment (v0vs) [inline src/strigo/index.ts:44]: I never used the native `fetch` yet, but similar to the example in the logger - should it check the `response.ok`?

File: services/logger.ts [hunk 6]
```diff
@@ -1,39 +1,82 @@
...
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${this.token}`
+      },
+      body: JSON.stringify({
+        severity,
+        message,
+        context,
+        dt: new Date().toISOString()
```
Comment (v0vs) [inline services/logger.ts:29]: Let's use something more standard like `timestamp` if we won't use the logtail

File: src/strigo/index.ts (hunk 5 above)
Comment (ZobelOmry) [reply src/strigo/index.ts:44]: yap

**Pull request merged**